end_to_end/eval_assert.py

```diff
@@ -44,6 +44,19 @@ def test_checkpointing(metrics_file, target):
     # step in saved checkpoint to loss of first step in restored checkpoint
     assert isclose(saved_loss, restored_loss, rel_tol=0.1)
 
+def test_determinism(metrics_file, target):
+  """Asserts over loss values from two runs"""
+  run_1 = 'run_1_' + metrics_file
+  run_2 = 'run_2_' + metrics_file
+
+  with open(run_1, 'r', encoding='utf8') as run_1_file,\
+    open(run_2, 'r', encoding='utf8') as run_2_file:
+    run_1_loss = json.loads(run_1_file.readlines()[-1])[target]
+    run_2_loss = json.loads(run_2_file.readlines()[-1])[target]
+    # Check that the two runs have the same loss
+    print(f"Run 1 loss:{run_1_loss}", flush=True)
+    print(f"Run 2 loss:{run_2_loss}", flush=True)
+    assert isclose(run_1_loss, run_2_loss, rel_tol=0.001)
```

Comment:
Just to help my understanding, what is the reason for allowing a (small?) mismatch tolerance in the loss check here for determinism?

I would have initially thought that the values would be equivalent?